src/models/state/wallet/mod.rs

```diff
@@ -49,14 +53,153 @@ pub const WALLET_DB_NAME: &str = "wallet";
 pub const WALLET_OUTPUT_COUNT_DB_NAME: &str = "wallout_output_count_db";
 
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
-struct SecretKeyMaterial(XFieldElement);
+pub struct SecretKeyMaterial(XFieldElement);
```

Comment:
Can we use `pub(crate)` here for `SecretKeyMaterial` and all its new associated functions?

Reply:
The problem is that they are being used from within `neptune-cli.rs`, which lives in its own crate and therefore does not have access to structs and functions marked `pub(crate)`.